### PR TITLE
Pset_EarthworksFillCommon

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_EarthworksFillCommon

## Description: 

Common properties for earthworks fill elements

| Property              | Datatype                | Description                                                  |
| --------------------- | ----------------------- | ------------------------------------------------------------ |
| SourceOfFill          | PEnum_SourceOfFill      | Source of fill                                               |
| SourceOfFillOther     | IfcLabel                | User defined Source of fill when SourceOfFill = OTHER        |
| CompactionMethod      | PEnum_CompactionMethod  | Method of compaction                                         |
| CompactionMethodOther | IfcLabel                | User defined method of compaction when CompactionMethod = OTHER |
| CompactionFactor      | IfcPositiveRatioMeasure | Reduction in volume on compaction                            |

## Enumeration: PEnum_SourceOfFill

| Constant  | Description |
| --------- | ----------- |
| IMPORTED  |The fill material is imported.|
| EXCAVATED |The fill material is excavated on site.|
| OTHER     |Some other source is used for the fill. Use property SourceOfFillOther to describe the source. |

## Enumeration: PEnum_CompactionMethod

| Constant | Description |
| -------- | ----------- |
| ROLLER   |Compaction using some type of roller equipment.|
| TAMPING  |Compaction using some kind of tamping equipment.|
| NONE     |No compaction.|
| OTHER    |Some other means of compaction. Use property CompactionMethodOther to describe the means.|

### CODE REVIEW

**Comment by SergejMuhic** [comment]
Punctuation, as a generic comment.

File: IFC4x3/Properties/s/SourceOfFill_1cA5Um37bAPRNMK84NGNf7/Documentation.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Source of fill
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SourceOfFill_1cA5Um37bAPRNMK84NGNf7/Documentation.md]:
```suggestion
Source of fill.
```

File: IFC4x3/PropertyEnumerations/c/PEnum_CompactionMethod.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/c/PEnum_CompactionMethod.xml:1]: Does this enumeration require any documentation?

File: IFC4x3/PropertyEnumerations/s/PEnum_SourceOfFill.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/s/PEnum_SourceOfFill.xml:1]: Any documentation required?

**Review by SergejMuhic — COMMENTED** [review]
Wondering whether the property enums should not have the common constants:
- `NOTKNOWN`
- `USERDEFINED`
- `UNSET`
instead of `OTHER`.

File: IFC4x3/PropertyEnumerations/c/PEnum_CompactionMethod.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/PropertyEnumerations/c/PEnum_CompactionMethod.xml:1]: Added in latest commit.

File: IFC4x3/PropertyEnumerations/s/PEnum_SourceOfFill.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/PropertyEnumerations/s/PEnum_SourceOfFill.xml:1]: Added in latest commit.

**Review by SergejMuhic — APPROVED** [review]

**Comment by SergejMuhic** [comment]
> Wondering whether the property enums should not have the common constants:
> 
> * `NOTKNOWN`
> * `USERDEFINED`
> * `UNSET`
>   instead of `OTHER`.

Are we sticking to `OTHER`?

**Comment by larswik** [comment]
> > Wondering whether the property enums should not have the common constants:
> > 
> > * `NOTKNOWN`
> > * `USERDEFINED`
> > * `UNSET`
> >   instead of `OTHER`.
> 
> Are we sticking to `OTHER`?

Fixed in commit [73559be](https://github.com/bSI-InfraRoom/IFC-Specification/pull/564/commits/73559bee734f272dd50d18a054af12ce8f2f4a8d).

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**